src/Symfony/Bundle/FrameworkBundle/Console/Application.php

```diff
@@ -40,7 +40,8 @@ public function __construct(KernelInterface $kernel)
         parent::__construct('Symfony', Kernel::VERSION);
 
         $inputDefinition = $this->getDefinition();
-        $inputDefinition->addOption(new InputOption('--env', '-e', InputOption::VALUE_REQUIRED, 'The Environment name.', $kernel->getEnvironment()));
+        $inputDefinition->addOption(new InputOption('--env', '-e', InputOption::VALUE_REQUIRED, 'The Environment name.', $kernel->getMode()));
+        $inputDefinition->addOption(new InputOption('--mode', null, InputOption::VALUE_REQUIRED, 'The Mode name.', $kernel->getMode()));
```

Comment:
this effectively turns `--mode` into a reserve option name. So there *is* a BC impact for that change